upgrades/upgrade_testing_framework/steps/step_create_source_snapshot.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+import upgrade_testing_framework.core.shell_interactions as shell
+
+class CreateSourceSnapshot(FrameworkStep):
+    """
+    This step creates the snapshot on the source cluster.
+    """
+
+    def _run(self):
+        """
+        The contents of this are quick/dirty.  Instead of raw curl commands we should probably be using the Python SDK
+        for Elasticsearch/OpenSearch.  At the very least, we should wrap the curl commands in our own pseudo-client.
+        """
+
+        # Get the state we need
+        shared_volume = self.state.shared_volume
+        source_cluster = self.state.source_cluster
+
+        # Begin the step body
+        self.logger.info("Creating snapshot of source cluster...")
+        port = source_cluster.rest_ports[0]
+
+        register_cmd = f"""curl -XPUT 'localhost:{port}/_snapshot/noldor' -H 'Content-Type: application/json' -d '{{
+            "type": "fs",
+                "settings": {{
+                    "location": "{shared_volume.mount_point}"
+            }}
+        }}'
+        """
```

Comment:
Would like to see at least a debug log for the name of the snapshot repo created and snapshot name later

Reply:
The `call_shell_command()` function logs all that.  That said - this `curl` code is going to be ripped out in the next few days/weeks anyways and replaced with "real" code, so...